node/src/reactor/queue_kind.rs

```diff
@@ -30,6 +30,18 @@ pub enum QueueKind {
     Api,
 }
 
+impl Display for QueueKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let str_value = match self {
+            QueueKind::NetworkIncoming => "NetworkIncoming",
+            QueueKind::Network => "Network",
+            QueueKind::Regular => "Regular",
+            QueueKind::Api => "Api",
+        };
+        write!(f, "{}", str_value.to_string())
```

Comment:
This allocates a string for no reason at at all,

```rust
write!(f, "{}", str_value)
```

should work here.

Reply:
cool, no idea why rustc complained then.

Reply:
An even better approach might be to 

```rust
match self {
            QueueKind::NetworkIncoming => write!(f, "{}", "NetworkIncoming"),
            //...
```

Reply:
Why is it better? I don't want to write `f.write_str` or `write!(f, ` multiple times. I will get rid of the allocation though.

Reply:
Naively (without optimization) it will not assign the value to a temporary variable before calling the function.

Well, because you asked: https://godbolt.org/z/xY6dhe

In debug mode, the `=> f.write_str("...")` version is 12 instructions and one jump, while the second one is 15 instructions with two jumps.

But let's core for your variant, since with `-C opt_level=3`, the compiler generates exactly the same output for both :)

Reply:
Done.